cmd/agent/app/reporter/grpc/collector_proxy.go

```diff
@@ -69,6 +69,5 @@ func (b ProxyBuilder) GetManager() configmanager.ClientConfigManager {
 
 // Close closes connections used by proxy.
 func (b ProxyBuilder) Close() error {
-	b.reporter.Close()
```

Comment:
why are we removing this?

Reply:
See, this is why I think reporters and handlers should be `io.Closer` as well even if most are no-op. We argued in a previous iteration that reporters are stateless and need no closing, but then, we need to remember to close this one specific reporter, which we only know that needs closing because a test fails in CI (passes locally, for some strange reason)... Without checking the struct's definition, we would even think that this could be any implementation of `Reporter`, but it's actually tied to a concrete reporter (`ClientMetricsReporter`).

Anyway, I restored this line.